sqlite_query.c

```diff
@@ -249,38 +574,67 @@ sqlite_bind_sql_var(Oid type, int attnum, Datum value, sqlite3_stmt * stmt, bool
 			}
 		case BYTEAOID:
 			{
-				int			len;
-				char	   *dat = NULL;
-				char	   *result = DatumGetPointer(value);
+				ret = sqlite_bind_blob_algo(attnum, value, stmt);
+				break;
+			}
+		case UUIDOID:
+			{
+				bool		uuid_as_blob = false;
 
-				if (VARATT_IS_1B(result))
+				if (relid)
 				{
-					len = VARSIZE_1B(result) - VARHDRSZ_SHORT;
-					dat = VARDATA_1B(result);
+					char * optv = get_column_option_string (relid, attnum, "column_type");
+					elog(DEBUG3, "sqlite_fdw : col type %s ", optv);
+					if (optv != NULL && strcmp_case_independend(optv, "BLOB") == 0)
+						uuid_as_blob = true;
 				}
-				else
+
+				if (uuid_as_blob)
 				{
-					len = VARSIZE_4B(result) - VARHDRSZ;
-					dat = VARDATA_4B(result);
+					unsigned char *dat = palloc0(UUID_LEN);		
+					pg_uuid_t* pg_uuid = DatumGetUUIDP(value);
+					elog(DEBUG2, "sqlite_fdw : bind uuid as blob");
+					memcpy(dat, pg_uuid->data, UUID_LEN);					
+					ret = sqlite3_bind_blob(stmt, attnum, dat, UUID_LEN, SQLITE_TRANSIENT);
+				}
+				else 
+				{
+					/* uuid as text */
+					char	   *outputString = NULL;
+					Oid			outputFunctionId = InvalidOid;
+					bool		typeVarLength = false;
+					getTypeOutputInfo(type, &outputFunctionId, &typeVarLength);
+					outputString = OidOutputFunctionCall(outputFunctionId, value); /* uuid text belongs to ASCII subset, no need to translate encoding */
+					ret = sqlite3_bind_text(stmt, attnum, outputString, -1, SQLITE_TRANSIENT);
 				}
-				ret = sqlite3_bind_blob(stmt, attnum, dat, len, SQLITE_TRANSIENT);
 				break;
 			}
-
 		default:
 			{
-				ereport(ERROR, (errcode(ERRCODE_FDW_INVALID_DATA_TYPE),
-								errmsg("cannot convert constant value to Sqlite value %u", type),
-								errhint("Constant value data type: %u", type)));
+				if (pgtypmod > 0)
+				{
+					const char	*pg_dataTypeName = TypeNameToString(makeTypeNameFromOid(type, pgtypmod));
```

Comment:
Could you explain about this error and the related testcase?

Reply:
This is error for unsupported PostgreSQL data type binding. I have meet this error for `uuid` data type before my implementation and improved this function to add human readable form of message if `pgtypmod` is passed in `bind...` function. No related testcases, this is _common diagnostics_, not someone for UUIDs only.

Reply:
Thank for explaining. I understood.

Reply:
Resolved.

Reply:
Added special testcase in https://github.com/pgspider/sqlite_fdw/pull/82/commits/fc1d6352183e7b8560e7f261f9dd23171ce26329, please confirm.

Reply:
Thank for your updating. I confirmed.

Reply:
Resolved again.